yt_dlp/extractor/archiveorg.py

```diff
@@ -285,38 +285,99 @@ class YoutubeWebArchiveIE(InfoExtractor):
             'info_dict': {
                 'id': '97t7Xj_iBv0',
                 'ext': 'mp4',
-                'title': '97t7Xj_iBv0'
+                'title': 'How Flexible Machines Could Save The World'
             }
-
         },
         {
-            # Video from 2012, webm format itag 45
+            # Video from 2012, webm format itag 45.
             'url': 'https://web.archive.org/web/20120712231619/http://www.youtube.com/watch?v=AkhihxRKcrs&gl=US&hl=en',
             'info_dict': {
                 'id': 'AkhihxRKcrs',
                 'ext': 'webm',
-                'title': 'AkhihxRKcrs'
+                'title': 'Limited Run: Mondo\'s Modern Classic 1 of 3 (SDCC 2012)'
             }
         },
         {
-            # Old flash-only video
+            # Old flash-only video. Webpage title starts with "YouTube - ".
             'url': 'https://web.archive.org/web/20081211103536/http://www.youtube.com/watch?v=jNQXAC9IVRw',
             'info_dict': {
                 'id': 'jNQXAC9IVRw',
                 'ext': 'unknown_video',
-                'title': 'jNQXAC9IVRw'
+                'title': 'Me at the zoo'
             }
+        },
+        {   # Some versions of Youtube have have "Youtube" as page title in html (and later rewritten by js).
+            'url': 'https://web.archive.org/web/http://www.youtube.com/watch?v=kH-G_aIBlFw',
+            'info_dict': {
+                'id': 'kH-G_aIBlFw',
+                'ext': 'mp4',
+                'title': 'kH-G_aIBlFw'
+            },
+            'expected_warnings': [
+                'Unable to extract video title',
+            ]
+        },
+        {
+            # First capture is a 302 redirect intermediary page.
+            'url': 'https://web.archive.org/web/20050214000000/http://www.youtube.com/watch?v=0altSZ96U4M',
+            'info_dict': {
+                'id': '0altSZ96U4M',
+                'ext': 'mp4',
+                'title': '0altSZ96U4M'
+            },
+            'expected_warnings': [
+                'Unable to extract video title',
+            ]
+        },
+        {
+            # Video not archived, only capture is unavailable video page
+            'url': 'https://web.archive.org/web/20210530071008/https://www.youtube.com/watch?v=lHJTf93HL1s&spfreload=10',
+            'only_matching': True,
         }
     ]
 
     def _real_extract(self, url):
         video_id = self._match_id(url)
+        title = video_id  # if we are not able get a title
+
+        def _extract_title(webpage):
+            page_title = self._html_search_regex(r'<title>([^<]*)</title>', webpage, 'title', fatal=False) or ''
+            page_title = page_title.strip()
+            # Unavailable video captures will only have youtube suffix/prefix as page title.
+            if page_title.endswith("- YouTube"):
```

Comment:
Issue: Sometimes there are more than 1 spaces separating the "-"
https://web.archive.org/web/20110712231407/http://www.youtube.com/watch?v=lTx3G6h2xyA